File: convert_string_to_camel_case.rb
def to_camel_case(str)
  words = str.include?('-') ? str.split('-') : str.split('_')
  cap_words = words.map(&:capitalize)
  cap_words.first.downcase! if ('a'..'z').include?(words[0][0])
  cap_words.join
end

p to_camel_case('the-stealth-warrior')
p to_camel_case('The_stealth_warrior')

File: go/convert_string_to_camel_case.go
package main

import (
	"fmt"
	"strings"
)

func ToCamelCase(s string) string {
	s = strings.ReplaceAll(s, "-", "_")
	words := strings.Split(s, "_")
	for i := 1; i < len(words); i++ {
		words[i] = strings.ToUpper(string(words[i][0])) + words[i][1:]
	}
	return strings.Join(words, "")
}

func main() {
	fmt.Println(ToCamelCase(""))
	fmt.Println(ToCamelCase("The_Stealth_Warrior"))
	fmt.Println(ToCamelCase("the-Stealth-Warrior"))
}
